src/disambiguities.jl

```diff
@@ -1,14 +1,100 @@
-Base.:(==)(::Missing, ::AbstractQuantity) = missing
-Base.isapprox(::AbstractQuantity, ::Missing; kws...) = missing
-Base.isapprox(::Missing, ::AbstractQuantity; kws...) = missing
+for op in (:isless, :(==), :isequal, :(<)), (type, _, _) in ABSTRACT_QUANTITY_TYPES
+    @eval begin
+        Base.$(op)(::$type, ::Missing) = missing
+        Base.$(op)(::Missing, ::$type) = missing
+    end
+end
+for op in (:isapprox,), (type, _, _) in ABSTRACT_QUANTITY_TYPES
+    @eval begin
+        Base.$(op)(::$type, ::Missing; kws...) = missing
+        Base.$(op)(::Missing, ::$type; kws...) = missing
+    end
+end
 
-Base.:(==)(::AbstractQuantity, ::WeakRef) = error("Cannot compare a quantity to a weakref")
-Base.:(==)(::WeakRef, ::AbstractQuantity) = error("Cannot compare a weakref to a quantity")
+for (type, _, _) in ABSTRACT_QUANTITY_TYPES
+    @eval begin
+        Base.:(==)(::$type, ::WeakRef) = error("Cannot compare a quantity to a weakref")
+        Base.:(==)(::WeakRef, ::$type) = error("Cannot compare a weakref to a quantity")
+    end
+end
 
 Base.:*(l::AbstractDimensions, r::Number) = error("Please use an `UnionAbstractQuantity` for multiplication. You used multiplication on types: $(typeof(l)) and $(typeof(r)).")
 Base.:*(l::Number, r::AbstractDimensions) = error("Please use an `UnionAbstractQuantity` for multiplication. You used multiplication on types: $(typeof(l)) and $(typeof(r)).")
 Base.:/(l::AbstractDimensions, r::Number) = error("Please use an `UnionAbstractQuantity` for division. You used division on types: $(typeof(l)) and $(typeof(r)).")
 Base.:/(l::Number, r::AbstractDimensions) = error("Please use an `UnionAbstractQuantity` for division. You used division on types: $(typeof(l)) and $(typeof(r)).")
+
+# Promotion ambiguities
+function Base.promote_rule(::Type{F}, ::Type{Bool}) where {F<:FixedRational}
+    return F
+end
+function Base.promote_rule(::Type{Bool}, ::Type{F}) where {F<:FixedRational}
+    return F
+end
+function Base.promote_rule(::Type{F}, ::Type{BigFloat}) where {F<:FixedRational}
+    return promote_type(Rational{eltype(F)}, BigFloat)
+end
+function Base.promote_rule(::Type{BigFloat}, ::Type{F}) where {F<:FixedRational}
+    return promote_type(Rational{eltype(F)}, BigFloat)
+end
+function Base.promote_rule(::Type{F}, ::Type{T}) where {F<:FixedRational,T<:AbstractIrrational}
+    return promote_type(Rational{eltype(F)}, T)
+end
+function Base.promote_rule(::Type{T}, ::Type{F}) where {F<:FixedRational,T<:AbstractIrrational}
+    return promote_type(Rational{eltype(F)}, T)
+end
+
+################################################################################
+# Assorted calls found by Aqua: ################################################
+################################################################################
+
+for type in (Signed, Float64, Float32, Rational), op in (:flipsign, :copysign)
+    @eval function Base.$(op)(x::$type, y::AbstractRealQuantity)
+        return $(op)(x, ustrip(y))
```

Comment:
```suggestion
        return Base.$(op)(x, ustrip(y))
```
for consistency?

Reply:
Nothing to address in this PR, but while I'm here: I think I pefer the style while `op` iterates over the function objects directly (e.g. `op in (Base.flipsign, Base.copysign)`). Dispatch is still possible without the symbol, e.g. see https://github.com/gaurav-arya/StochasticAD.jl/blob/3638b3b36450451e07f890c033fb253199ca83b5/src/general_rules.jl#L200-L201.

But the iteration over symbol approach like here indeed is very common. It just seems a bit less elegant, e.g. what if you wanted `op` to iterate over multiple methods owned by different modular?

Reply:
Thanks. I'm open to it, I guess we could just do this in another PR?